### DISCUSSION TITLE
AWS ECR Scan Vulnerability Failed RHSA-2024:0267

### DISCUSSION THREAD

**Discussion by wynnteo** [quay/quay#2615] [Q&A] [answered]
Got this vulnerability scan HIGH 

The java-17-openjdk packages provide the OpenJDK 17 Java Runtime Environment and the OpenJDK 17 Java Software Development Kit. Security Fix(es): * OpenJDK: array out-of-bounds access due to missing range check in C1 compiler (8314468) (CVE-2024-20918) * OpenJDK: incorrect handling of ZIP files with duplicate entries (8276123) (CVE-2024-20932) * OpenJDK: RSA padding issue and timing side-channel attack against TLS (8317547) (CVE-2024-20952) * OpenJDK: JVM class file verifier flaw allows unverified bytecode execution (8314295) (CVE-2024-20919) * OpenJDK: range check loop optimization issue (8314307) (CVE-2024-20921) * OpenJDK: logging of digital signature private keys (8316976) (CVE-2024-20945) For more details about the security issue(s), including the impact, a CVSS score, acknowledgments, and other related information, refer to the CVE page(s) listed in the References section. Bug Fix(es): * When Transparent Huge Pages (THP) are unconditionally enabled on a system, Java applications using many threads were found to have a large Resident Set Size (RSS). This was due to a race between the kernel transforming thread stack memory into huge pages and the Java Virtual Machine (JVM) shattering these pages into smaller ones when adding a guard page. This release resolves this issue by getting glibc to insert a guard page and prevent the creation of huge pages. (RHEL-13930, RHEL-13931, RHEL-13934, RHEL-13935)

Any idea how to fix?

In my Dockerfile this is the version I using
FROM quay.io/keycloak/keycloak:nightly as builder

**Comment by ibazulic** [comment, marked as answer]
@wynnteo Hi there! Quay team is responsible for Quay.io as a service, the maintaining and preserving registry operations and registry functionality, we are not responsible for individual images stored on the registry.  If you have a problem with a particular image (in this case Keycloak), I would recommend you contact maintainers of that image to see if a fix for the detected vulnerabilities is available.
